Fix release script and documentation

mkuznyetsov opened the pull request:
Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Fix release script using improper variable.
Remove old documentation in RELEASE.md

### What issues does this PR fix or reference?


### Specify the version of the product this PR applies to.


### PR Checklist

As the author of this Pull Request I made sure that:

- [ ] `vale` has been run successfully against the PR branch
- [ ] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

mkuznyetsov committed:
Fix release script and documentation

Signed-off-by: Mykhailo Kuznietsov <mkuznets@redhat.com>

nickboldt commented on make-release.sh:
```diff
@@ -39,7 +39,7 @@ if [[ ! ${VERSION} ]]; then
 fi
 
-if [[ ${USE_TMP_DIR} ]]; then
+if [[ ${USE_TMP_DIR} -eq 1 ]]; then
```
this will never be true. Check out line 18, where you're changing it from 0 to 0 :|

nickboldt committed:
Update make-release.sh

 '-tmp' and  '--use-tmp-dir' flags shoulld set USE_TMP_DIR=1, or the if-checks will never be true :D
